packages/fleet/lib/models/node_config_overrides.ts

```diff
@@ -0,0 +1,153 @@
+        }
+        return Ok(DBNodeConfigOverride.from(created));
+    } catch (e) {
+        return Err(new FleetError('node_config_override_creation_failed', { cause: e, context: props }));
+    }
+}
+
+export async function update(
+    db: knex.Knex,
+    props: Partial<NodeConfigOverride> & Required<Pick<NodeConfigOverride, 'routingId'>>
+): Promise<Result<NodeConfigOverride, FleetError>> {
+    try {
+        const now = new Date();
+        const toUpdate: Partial<DBNodeConfigOverride> = {
+            ...(props.image ? { image: props.image } : {}),
+            ...(props.cpuMilli ? { cpu_milli: props.cpuMilli } : {}),
+            ...(props.memoryMb ? { memory_mb: props.memoryMb } : {}),
+            ...(props.storageMb ? { storage_mb: props.storageMb } : {}),
+            updated_at: now
+        };
+        const [updated] = await db
+            .from<DBNodeConfigOverride>(NODE_CONFIG_OVERRIDES_TABLE)
+            .update(toUpdate)
+            .where({ routing_id: props.routingId })
+            .returning('*');
+        if (!updated) {
+            return Err(new FleetError('node_config_override_update_failed', { context: props }));
+        }
+        return Ok(DBNodeConfigOverride.from(updated));
+    } catch (e) {
+        return Err(new FleetError('node_config_override_update_failed', { cause: e, context: props }));
+    }
+}
+
+export async function remove(db: knex.Knex, routingId: RoutingId): Promise<Result<NodeConfigOverride, FleetError>> {
+    try {
+        const [deleted] = await db.from<DBNodeConfigOverride>(NODE_CONFIG_OVERRIDES_TABLE).delete().where({ routing_id: routingId }).returning('*');
+        if (!deleted) {
+            return Err(new FleetError('node_config_override_not_found', { context: { routingId } }));
+        }
+        return Ok(DBNodeConfigOverride.from(deleted));
+    } catch (e) {
+        return Err(new FleetError('node_config_override_delete_failed', { cause: e, context: { routingId } }));
+    }
+}
+
+export async function search(
+    db: knex.Knex,
+    params: {
+        routingIds?: RoutingId[];
+    }
+): Promise<Result<Map<RoutingId, NodeConfigOverride>, FleetError>> {
+    try {
+        const query = db.from<DBNodeConfigOverride>(NODE_CONFIG_OVERRIDES_TABLE).select('*');
+        if (params.routingIds) {
+            query.whereIn('routing_id', params.routingIds);
+        }
+        const nodeConfigOverrides = await query;
+        const nodeConfigOverridesMap = new Map<RoutingId, NodeConfigOverride>();
+        for (const nodeConfigOverride of nodeConfigOverrides) {
+            nodeConfigOverridesMap.set(nodeConfigOverride.routing_id, DBNodeConfigOverride.from(nodeConfigOverride));
+        }
+        return Ok(nodeConfigOverridesMap);
+    } catch (e) {
+        return Err(new FleetError('node_config_override_search_failed', { cause: e, context: { params } }));
+    }
+}
+
+export async function setForAll(
```

Comment:
imo Seems like a footgun to be able to update more than just the image

Reply:
I think it could be handy but I see your point. I will change